diff -rup /var/tmp/portage/sys-apps/exynos5-hwcomposer-27112013/work/exynos5-hwcomposer-27112013/tools/hardkernel/exynos5-hwcomposer/include/log.h /mnt/external_sd/exynos5-hwcomposer-27112013-new/tools/hardkernel/exynos5-hwcomposer/include/log.h
--- /var/tmp/portage/sys-apps/exynos5-hwcomposer-27112013/work/exynos5-hwcomposer-27112013/tools/hardkernel/exynos5-hwcomposer/include/log.h	2014-06-26 12:29:23.706256329 +0100
+++ /mnt/external_sd/exynos5-hwcomposer-27112013-new/tools/hardkernel/exynos5-hwcomposer/include/log.h	2014-06-26 12:05:02.503282992 +0100
@@ -7,26 +7,53 @@
 extern "C" {
 #endif
 
-#if 1
+#define LOG_OUT_SYSLOG
+//#define LOG_OUT_FPRINTF
+
+#ifdef LOG_OUT_SYSLOG
+#include <syslog.h>
+
+/*
+#define ALOG(level,format, ...) \
+do { \
+syslog(level, format); \
+} while(0)
+#define ALOGD(format, ...)      ALOG(LOG_DEBUG,"%s,%d -- " format "\n", __FUNCTION__, __LINE__, ##__VA_ARGS__)
+
+*/
+
+#define ALOG(level,format, ...) \
+        do { \
+		syslog(level, "%s,%d -- " format "\n", __FUNCTION__, __LINE__, ##__VA_ARGS__); \
+	} while(0)
+
+#define ALOGD(format, ...)	ALOG(LOG_DEBUG, format, ##__VA_ARGS__)
+#define ALOGV(format, ...)	ALOG(LOG_NOTICE, format, ##__VA_ARGS__)
+#define ALOGI(format, ...)	ALOG(LOG_INFO, format, ##__VA_ARGS__)
+#define ALOGE(format, ...)	ALOG(LOG_ERR, format, ##__VA_ARGS__)
+#define ALOGW(format, ...)	ALOG(LOG_WARNING, format, ##__VA_ARGS__)
+#define ALOGF(format, ...)	ALOG(LOG_ALERT, format, ##__VA_ARGS__)
+
+#else
+#ifdef LOG_OUT_FPRINTF
 #define ALOGD(format, ...) \
 	do { \
 		fprintf(stderr, "%s,%d -- " format "\n", \
 				__FUNCTION__, __LINE__, ##__VA_ARGS__); \
 	} while(0)
+#else
+#define ALOGD
+#endif
+
 
 #define ALOGV	ALOGD
 #define ALOGI	ALOGD
 #define ALOGE	ALOGD
 #define ALOGW	ALOGD
 #define ALOGF	ALOGD
-#else
-#define ALOGD
-#define ALOGV
-#define ALOGI
-#define ALOGE
-#define ALOGW
 #endif
 
+
 #if __cplusplus
 };
 #endif
diff -rup /var/tmp/portage/sys-apps/exynos5-hwcomposer-27112013/work/exynos5-hwcomposer-27112013/tools/hardkernel/exynos5-hwcomposer/src/hardware/samsung_slsi/exynos/libv4l2/exynos_subdev.c /mnt/external_sd/exynos5-hwcomposer-27112013-new/tools/hardkernel/exynos5-hwcomposer/src/hardware/samsung_slsi/exynos/libv4l2/exynos_subdev.c
--- /var/tmp/portage/sys-apps/exynos5-hwcomposer-27112013/work/exynos5-hwcomposer-27112013/tools/hardkernel/exynos5-hwcomposer/src/hardware/samsung_slsi/exynos/libv4l2/exynos_subdev.c	2014-06-26 12:29:23.736255873 +0100
+++ /mnt/external_sd/exynos5-hwcomposer-27112013-new/tools/hardkernel/exynos5-hwcomposer/src/hardware/samsung_slsi/exynos/libv4l2/exynos_subdev.c	2014-06-26 12:05:02.558283153 +0100
@@ -44,6 +44,8 @@
 #include <log.h>
 #endif
 
+#include <string.h>
+
 #define SUBDEV_MINOR_MAX 191
 
 static int __subdev_open(const char *filename, int oflag, va_list ap)
diff -rup /var/tmp/portage/sys-apps/exynos5-hwcomposer-27112013/work/exynos5-hwcomposer-27112013/tools/hardkernel/exynos5-hwcomposer/src/hardware/samsung_slsi/exynos/libv4l2/exynos_v4l2.c /mnt/external_sd/exynos5-hwcomposer-27112013-new/tools/hardkernel/exynos5-hwcomposer/src/hardware/samsung_slsi/exynos/libv4l2/exynos_v4l2.c
--- /var/tmp/portage/sys-apps/exynos5-hwcomposer-27112013/work/exynos5-hwcomposer-27112013/tools/hardkernel/exynos5-hwcomposer/src/hardware/samsung_slsi/exynos/libv4l2/exynos_v4l2.c	2014-06-26 12:29:23.736255873 +0100
+++ /mnt/external_sd/exynos5-hwcomposer-27112013-new/tools/hardkernel/exynos5-hwcomposer/src/hardware/samsung_slsi/exynos/libv4l2/exynos_v4l2.c	2014-06-26 12:05:03.498285871 +0100
@@ -47,6 +47,8 @@
 #include <log.h>
 #endif
 
+#include <string.h>
+
 #define VIDEODEV_MINOR_MAX 63
 
 //#define EXYNOS_V4L2_TRACE 0
diff -rup /var/tmp/portage/sys-apps/exynos5-hwcomposer-27112013/work/exynos5-hwcomposer-27112013/tools/hardkernel/exynos5-hwcomposer/src/system/core/libsync/sync.c /mnt/external_sd/exynos5-hwcomposer-27112013-new/tools/hardkernel/exynos5-hwcomposer/src/system/core/libsync/sync.c
--- /var/tmp/portage/sys-apps/exynos5-hwcomposer-27112013/work/exynos5-hwcomposer-27112013/tools/hardkernel/exynos5-hwcomposer/src/system/core/libsync/sync.c	2014-06-26 12:29:23.746255721 +0100
+++ /mnt/external_sd/exynos5-hwcomposer-27112013-new/tools/hardkernel/exynos5-hwcomposer/src/system/core/libsync/sync.c	2014-06-26 12:20:37.823326102 +0100
@@ -27,6 +27,8 @@
 #include <stdint.h>
 #endif
 
+#include <stdlib.h>
+
 #include <sys/ioctl.h>
 #include <sys/stat.h>
 #include <sys/types.h>
diff -rup /var/tmp/portage/sys-apps/exynos5-hwcomposer-27112013/work/exynos5-hwcomposer-27112013/tools/hardkernel/exynos5-hwcomposer/tools/exynos5-hwcomposer.cpp /mnt/external_sd/exynos5-hwcomposer-27112013-new/tools/hardkernel/exynos5-hwcomposer/tools/exynos5-hwcomposer.cpp
--- /var/tmp/portage/sys-apps/exynos5-hwcomposer-27112013/work/exynos5-hwcomposer-27112013/tools/hardkernel/exynos5-hwcomposer/tools/exynos5-hwcomposer.cpp	2014-06-26 12:29:23.746255721 +0100
+++ /mnt/external_sd/exynos5-hwcomposer-27112013-new/tools/hardkernel/exynos5-hwcomposer/tools/exynos5-hwcomposer.cpp	2014-06-26 12:05:02.528283065 +0100
@@ -70,7 +70,7 @@ static void exynos5_cleanup_gsc_m2m(exyn
 
 static void dump_handle(private_handle_t *h)
 {
-	printf(" * fd = %d, format = %d, width = %u, height = %u,"
+	ALOGD(" * fd = %d, format = %d, width = %u, height = %u,"
 			"stride = %u, vstride = %u\n",
 			h->fd, h->format, h->width, h->height,
 			h->stride, h->vstride);
@@ -78,7 +78,7 @@ static void dump_handle(private_handle_t
 
 static void dump_layer(hwc_layer_1_t const *l)
 {
-	printf("layer type=%d, flags=%08x, handle=%p, tr=%02x, blend=%04x, "
+	ALOGD("layer type=%d, flags=%08x, handle=%p, tr=%02x, blend=%04x, "
 			"{%d,%d,%d,%d}, {%d,%d,%d,%d}\n",
 			l->compositionType, l->flags, l->handle, l->transform,
 			l->blending,
@@ -97,11 +97,11 @@ static void dump_layer(hwc_layer_1_t con
 
 static void dump_config(int i, s3c_fb_win_config &c)
 {
-	printf("window %u configuration:", i);
-	printf("\tstate = %u\n", c.state);
+	ALOGD("window %u configuration:", i);
+	ALOGD("\tstate = %u\n", c.state);
 
 	if (c.state == c.S3C_FB_WIN_STATE_BUFFER) {
-		printf("\t\tfd = %d, offset = %u, stride = %u, "
+		ALOGD("\t\tfd = %d, offset = %u, stride = %u, "
 				"x = %d, y = %d, w = %u, h = %u, "
 				"format = %u, blending = %u\n",
 				c.fd, c.offset, c.stride,
@@ -109,20 +109,20 @@ static void dump_config(int i, s3c_fb_wi
 				c.format, c.blending);
 	}
 	else if (c.state == c.S3C_FB_WIN_STATE_COLOR) {
-		printf("\t\tcolor = %u\n", c.color);
+		ALOGD("\t\tcolor = %u\n", c.color);
 	}
 }
 
 static void dump_gsc_img(exynos_gsc_img &c)
 {
-	printf("\tx = %u, y = %u, w = %u, h = %u, fw = %u, fh = %u\n",
+	ALOGD("\tx = %u, y = %u, w = %u, h = %u, fw = %u, fh = %u\n",
 			c.x, c.y, c.w, c.h, c.fw, c.fh);
-	printf("\tf = %u", c.format);
-	printf("\taddr = {%u, %u, %u}, rot = %u,"
+	ALOGD("\tf = %u", c.format);
+	ALOGD("\taddr = {%u, %u, %u}, rot = %u,"
 			"cacheable = %u, drmMode = %u\n",
 			c.yaddr, c.uaddr, c.vaddr, c.rot,
 			c.cacheable, c.drmMode);
-	printf("\tnarrowRgb = %u, acquireFenceFd = %d,"
+	ALOGD("\tnarrowRgb = %u, acquireFenceFd = %d,"
 			"releaseFenceFd = %d, mem_type = %u\n",
 			c.narrowRgb, c.acquireFenceFd,
 			c.releaseFenceFd, c.mem_type);
@@ -520,7 +520,7 @@ bool hdmi_is_preset_supported(struct exy
             return -1;
         }
 
-		printf("supported preset %d: preset=%02d"
+		ALOGD("supported preset %d: preset=%02d"
 				"width=%d height=%d name=%s\n",
 				enum_preset.index, enum_preset.preset,
 				enum_preset.width, enum_preset.height,
@@ -2342,7 +2342,7 @@ static int exynos5_prepare_hdmi(exynos5_
 static int exynos5_prepare_wfd(exynos5_hwc_composer_device_1_t *pdev,
         hwc_display_contents_1_t* contents)
 {
-    ALOGV("preparing %u layers for WFD", contents->numHwLayers);
+    ALOGV("preparing %lu layers for WFD", contents->numHwLayers);
     hwc_layer_1_t *video_layer = NULL;
 
     for (size_t i = 0; i < contents->numHwLayers; i++) {
@@ -3296,7 +3296,7 @@ static int exynos5_set_hdmi(exynos5_hwc_
         hwc_layer_1_t &layer = contents->hwLayers[i];
 
         if (layer.flags & HWC_SKIP_LAYER) {
-            ALOGV("HDMI skipping layer %d", i);
+            ALOGV("HDMI skipping layer %u", i);
             continue;
         }
 
@@ -4045,10 +4045,10 @@ static int exynos5_blank(struct hwc_comp
 
 static void exynos5_dump(exynos5_hwc_composer_device_1_t* pdev)
 {
-    printf("  hdmi_enabled=%u\n", pdev->hdmi_enabled);
+    ALOGD("  hdmi_enabled=%u\n", pdev->hdmi_enabled);
     if (pdev->hdmi_enabled)
-        printf("    w=%u, h=%u\n", pdev->hdmi_w, pdev->hdmi_h);
-    printf(
+        ALOGD("    w=%u, h=%u\n", pdev->hdmi_w, pdev->hdmi_h);
+    ALOGD(
             "   type   |  handle  |  color   | blend | format |   position    |     size      | gsc \n"
             "----------+----------|----------+-------+--------+---------------+---------------------\n");
     //        8_______ | 8_______ | 8_______ | 5____ | 6_____ | [5____,5____] | [5____,5____] | 3__ \n"
@@ -4056,33 +4056,33 @@ static void exynos5_dump(exynos5_hwc_com
     for (size_t i = 0; i < NUM_HW_WINDOWS; i++) {
         struct s3c_fb_win_config &config = pdev->last_config[i];
         if (config.state == config.S3C_FB_WIN_STATE_DISABLED) {
-            printf(" %8s | %8s | %8s | %5s | %6s | %13s | %13s",
+            ALOGD(" %8s | %8s | %8s | %5s | %6s | %13s | %13s",
                     "OVERLAY", "-", "-", "-", "-", "-", "-");
         }
         else {
             if (config.state == config.S3C_FB_WIN_STATE_COLOR)
-                printf(" %8s | %8s | %8x | %5s | %6s", "COLOR",
+                ALOGD(" %8s | %8s | %8x | %5s | %6s", "COLOR",
                         "-", config.color, "-", "-");
             else
-                printf(" %8s | %8x | %8s | %5x | %6x",
+                ALOGD(" %8s | %8u | %8s | %5x | %6x",
                         pdev->last_fb_window == i ? "FB" : "OVERLAY",
                         intptr_t(pdev->last_handles[i]),
                         "-", config.blending, config.format);
 
-            printf(" | [%5d,%5d] | [%5u,%5u]", config.x, config.y,
+            ALOGD(" | [%5d,%5d] | [%5u,%5u]", config.x, config.y,
                     config.w, config.h);
         }
         if (pdev->last_gsc_map[i].mode == exynos5_gsc_map_t::GSC_NONE) {
-            printf(" | %3s", "-");
+            ALOGD(" | %3s", "-");
         } else {
-            printf(" | %3d",
+            ALOGD(" | %3d",
                     AVAILABLE_GSC_UNITS[pdev->last_gsc_map[i].idx]);
             if (pdev->last_gsc_map[i].mode == exynos5_gsc_map_t::GSC_M2M)
-                printf(" | %10s(%dtimes)","GSC_M2M", pdev->need_gsc_op_twice ? 2 : 1);
+                ALOGD(" | %10s(%dtimes)","GSC_M2M", pdev->need_gsc_op_twice ? 2 : 1);
             else
-                printf(" | %10s","GSC_LOCAL");
+                ALOGD(" | %10s","GSC_LOCAL");
         }
-	printf("\n");
+	ALOGD("\n");
     }
 }
 
@@ -4478,7 +4478,7 @@ static int createDisplay(struct exynos5_
 		case HWC_DISPLAY_DPI_Y:
 			break;
 		default:
-			ALOGE("unknown display attribute[%d] %#x",
+			ALOGE("unknown display attribute[%u] %#x",
 					i, display_attributes[i]);
 			break;
 		}
@@ -4544,7 +4544,7 @@ int	main(void)
 	if(composer->hdmi_hpd)  init_composer(composer);
 
 	while (1) {
-//		printf("dennis -- ???\n");
+//		ALOGD("dennis -- ???\n");
 #if 0
 		usleep(500);
 #else
